Source/DafnyCore/Resolver/PreType/PreTypeResolve.Expressions.cs

```diff
@@ -1534,6 +1534,15 @@ public Expression ResolveNameSegment(NameSegment expr, bool isLastNameSegment, L
             }
             // nevertheless, set "receiver" to a value so we can continue resolution
           }
+          if (resolutionContext.InInvariant && member is Field && currentClass.InheritedMembers.Contains(member)) {
+            if (complain) {
```

Comment:
Off-topic: I'm amused by the idea that if `complain` is false, it could still raise an error but be more polite about it. :)